tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -186,25 +187,47 @@ def test_brand_string(test_microvm_with_api, network_config):
     "0xc0000083",  # MSR_CSTAR
     "0xc0000100",  # MSR_FS_BASE
     "0xc0000101",  # MSR_GS_BASE
+    # MSRs below are required on T2A but we
+    # we are adding them to the common exception list to keep things simple
+    "0x834"     ,  # LVT Performance Monitor Interrupt Register.
+    "0xc0010007",  # MSR_K7_PERFCTR3
+    "0xc001020b",  # Performance Event Counter MSR_F15H_PERF_CTR5
+    "0xc0011029",  # MSR_F10H_DECFG also referred to as MSR_AMD64_DE_CFG
 ]
+
+# CPU templates which support MSR test
+MSR_SUPPORTED_TEMPL = ["T2S", "T2A"]
+MSR_SUPPORTED_VENDORS = [
+                            cpuid_utils.CpuVendor.INTEL,
+                            cpuid_utils.CpuVendor.AMD
+                        ]
 # fmt: on
 
 
+@pytest.fixture(
+    name="msr_cpu_template",
+    params=set(SUPPORTED_CPU_TEMPLATES).intersection(MSR_SUPPORTED_TEMPL),
+)
+def msr_cpu_template_fxt(request):
+    """CPU template fixture for MSR read/write supported cpu templates"""
+    return request.param
+
+
 @pytest.mark.skipif(
     PLATFORM != "x86_64", reason="CPU features are masked only on x86_64."
 )
 @pytest.mark.skipif(
-    cpuid_utils.get_cpu_vendor() != cpuid_utils.CpuVendor.INTEL,
-    reason="CPU templates are only available on Intel.",
+    cpuid_utils.get_cpu_vendor() not in MSR_SUPPORTED_VENDORS,
```

Comment:
I don't think this is needed as you are already filtering based on templates.

Reply:
done.